src/views/Submit.vue

```diff
@@ -145,6 +145,21 @@ export default {
 
 	mixins: [ViewsMixin],
 
+	/*
+     * This is used to confirm that the user wants to leave the page
+     * if the form is unsubmitted.
+     */
+	beforeRouteUpdate(to, from, next) {
+		// beforeRouteLeave is called when the static route changes.
+		// Cancel the navigation, if the form has unsaved edits and the user did not
+		// confirm leave. This prevents accidentally losing changes
+		if (this.confirmLeaveForm()) {
+			next()
+		}
+
+		return false
```

Comment:
This seems to be wrong, in the unload event listener you have:
`if (confirmLeaveForm) then cancel`

But here it is reversed, I would stick with this logic here as "confirmLeave" sounds like it is confirmed to leave if true, so if true -> leave. Otherwise cancel.

Also we still use vue-router 3 so the syntax should be:
```suggestion
		if (this.confirmLeaveForm()) {
			next()
		}

		next(false)
```

Reply:
Did you already consider reversing the logic in this suggestion? Or shouldn't it be !this.confirmLeaveForm() then?

Reply:
All suggestions were based on the reversed logic, meaning confirmLeaveForm return true if you should continue with the navigation.

So here is no negation needed